source/kraken/tests/apply_disruption_test.cpp

```diff
@@ -109,3 +110,105 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
+ *
+ *
+ *                            closed         closed
+ *                            [10-12]        [11-14]
+ *                S1 ---------- S2 ------------S3
+ *
+ * vj1:           8h            9h             10h
+ * no vj1_adated
+ *
+ * vj2:           10h           10h30          11h
+ * vj2_adated:    10h           X              X         (Note for XL: the best would be to clean those vj, but as long as it does not code dump, it's not important not to clean them)
+ *                                                            --> ie either do a routing test with vj2/vj3 or clean them
+ * vj3:           10h                          11h
+ * vj3_adapted:   10h                          X
+ *
+ * vj4:           8h            10h            10h30
+ * vj4_adapted:   8h            X              10h30
+ *
+ * vj5:           8h            13h            13h30
+ * vj5_adapted:   8h            13h            X
+ *
+ */
+BOOST_AUTO_TEST_CASE(multiple_impact_on_stops_different_hours) {
+    ed::builder b("20160101");
+
+    b.sa("S1")("S1");
+    b.sa("S2")("S2");
+    b.sa("S2")("S2");
+
+    const auto* vj1 = b.vj("A").uri("vj1")("S1", "08:00"_t)("S2", "09:00"_t)("S3", "10:00"_t).make();
+    const auto* vj2 = b.vj("A").uri("vj2")("S1", "10:00"_t)("S2", "10:30"_t)("S3", "11:00"_t).make();
+    const auto* vj3 = b.vj("A").uri("vj3")("S1", "10:00"_t)                 ("S3", "10:00"_t).make();
+    const auto* vj4 = b.vj("A").uri("vj4")("S1", "08:00"_t)("S2", "10:00"_t)("S3", "10:30"_t).make();
+    const auto* vj5 = b.vj("A").uri("vj5")("S1", "08:00"_t)("S2", "13:00"_t)("S3", "13:30"_t).make();
+
+    b.finish();
+    b.data->pt_data->index();
+    b.data->build_raptor();
+    b.data->build_uri();
+
+    auto* s1 = b.data->pt_data->stop_points_map["S1"];
+    auto* s2 = b.data->pt_data->stop_points_map["S2"];
+    auto* s3 = b.data->pt_data->stop_points_map["S3"];
+
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 5);
+
+    navitia::apply_disruption(b.impact(nt::RTLevel::Adapted, "S2_closed")
+                              .severity(nt::disruption::Effect::NO_SERVICE)
+                              .on(nt::Type_e::StopPoint, "S2")
+                              .application_periods(btp("20160101T100000"_dt, "20160101T1120001"_dt))
+                              .get_disruption(),
+                              *b.data->pt_data, *b.data->meta);
+
+    navitia::apply_disruption(b.impact(nt::RTLevel::Adapted, "S3_closed")
+                              .severity(nt::disruption::Effect::NO_SERVICE)
+                              .on(nt::Type_e::StopPoint, "S3")
+                              .application_periods(btp("20160101T110000"_dt, "20160101T1140001"_dt))
+                              .get_disruption(),
+                              *b.data->pt_data, *b.data->meta);
+
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 10); //XL: or 8 if you clean vj2_adapted and vj3_adapted
```

Comment:
I don't get it why 10, I got 8 vjs even though I don't clean up... Let's talk about when i'm back :)